common/v2/features/ProtectTransaction/txStateFactory.tsx

```diff
@@ -0,0 +1,188 @@
+import { useCallback, useContext } from 'react';
+import { Arrayish, hexlify } from 'ethers/utils';
+
+import { fromTxReceiptObj, makeTxConfigFromSignedTx, TUseStateReducerFactory } from 'v2/utils';
+import { AssetContext } from 'v2/services/Store/Asset';
+import {
+  IFormikFields,
+  ISignedTx,
+  ITxConfig,
+  ITxObject,
+  ITxReceipt,
+  Asset,
+  ITxStatus
+} from 'v2/types';
+import { hexWeiToString } from 'v2/services/EthService/utils';
+import { processFormDataToTx } from 'v2/features/SendAssets/helpers';
+import { getBaseAssetByNetwork, NetworkContext } from 'v2/services/Store/Network';
+import { ProviderHandler } from 'v2/services/EthService/network';
+import { AccountContext } from 'v2/services/Store/Account';
+import { getNonce } from 'v2/services/EthService';
+import { PROTECTED_TX_FEE_ADDRESS } from 'v2/config';
+import { StoreContext } from 'v2/services/Store';
+
+const protectedTxConfigInitialState = {
+  tx: {
+    gasLimit: null,
+    gasPrice: null,
+    nonce: null,
+    data: null,
+    to: null
+  },
+  amount: null,
+  receiverAddress: null,
+  senderAccount: null,
+  network: undefined,
+  asset: null
+};
+
+interface State {
+  txConfig: ITxConfig;
+  txReceipt?: ITxReceipt;
+  signedTx: ISignedTx; // make sure signedTx is only used within stateFactory
+}
+
+const ProtectedTxConfigFactory: TUseStateReducerFactory<State> = ({ state, setState }) => {
```

Comment:
We now have a custom hook `useTxMulti` that can manage `gasEstimates`, sending transactions and updating tx status eg. BROADCAST, CONFIRMED etc. It should be able to replace most of this code.